engine/classes/ElggVolatileMetadataCache.php

```diff
@@ -289,7 +289,7 @@ public function populateFromEntities($guids) {
 				"JOIN {$db_prefix}metastrings n ON n_table.name_id = n.id",
 			),
 			'selects' => array('n.string AS name', 'v.string AS value'),
-			'order_by' => 'n_table.entity_guid, n_table.time_created ASC',
+			'order_by' => 'n_table.entity_guid, n_table.id ASC',
```

Comment:
only custom queries can update time_created, but it "could" happen, and i believe there is no real downside in ordering by time created first and then by id, and it also solves your issue (= correct order for the same time_created)